### PR TITLE
Prep azure-search-documents for Patch Release

### LINKED ISSUE #23406: Align dependencies for azure-search-documents
There are dependency conflicts in azure-search-documents with respect to azure-core. Please fix them as part of the release.

image

You can also find the latest report at https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-java/staging/dependency_conflictlist.html which you may want to glance once azure-core releases.

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes #23406

### CODE REVIEW

File: sdk/search/azure-search-documents/README.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ Use the Azure Cognitive Search client library to:
 <dependency>
     <groupId>com.azure</groupId>
     <artifactId>azure-search-documents</artifactId>
-    <version>11.4.0</version>
```
Comment (pallavit) [inline sdk/search/azure-search-documents/README.md:41]: Would we not need to update the POM file?

Comment (alzimmermsft) [reply sdk/search/azure-search-documents/README.md:41]: Yep, forgot it

**Review by pallavit — APPROVED** [review]
:shipit:

**Review by joshfree — APPROVED** [review]

**Comment by joshfree** [comment]
Thanks @alzimmermsft

**Pull request merged**